### ISSUE TITLE
Github Actions - CI and tests

### ISSUE THREAD

**Issue by jbrezmorf** [flow123d/flow123d#1467]
After the multijob plugin of the Jenkins became obsolete, we were forced to basically rewrite the whole setup with the usage of the Jenkins pipeline plugin. It seems simpler to migrate the whole CI to Github actions which are naturally configured by the code, and we need not care about the hardware.
There are some issues thou:

- [x] continue-on-error for a step or job makes upper unit (job or workflow) succeed (since errors of subunits are ignored), 
  this is problematic especially for the steps, since these are visible only after diving into the job. 
  Resolution: use jobs for individual tests and implement a non-continue-on-error conclusion job collecting the results of the 
  matrix of the test jobs. Pass the build directory through the artifacts.

- [x]  For the release package distinguish the continuous build to the ci-<environment> image and release build to the flow123d-<environment> image the second may also include tag creation and should be triggered just manually via workflow-dispatch.

- [x] Fix the published images with wrong version tag in flow123d-gnu.

- [x] publishing the to astra through the SCP takes over 30 minutes because publishing to many small files in the HTML doc. 
  TODO: package into a tar, transfer and unpack through SSH call 

- [ ] move 'bin/set_variables.sh' to root of the every development image (empty for gnu) source as part of the make
- [ ] #1535

- [ ] (student) Study and propose usage of Github Checks API (can at least provide specific information to pulls)

- [ ] (student) Pull request event specific actions (e.g. performance and release tests before the merge)

- [ ] (student) Postpone performance tests (and package builds) to night.

**Comment by jbrezmorf** [comment]
Essential things done, further improvment ides summarized in : #1536
